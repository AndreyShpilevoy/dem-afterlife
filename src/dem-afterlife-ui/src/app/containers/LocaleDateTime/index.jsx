/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-mutation: 0, fp/no-this: 0 */

import React, {PureComponent} from 'react';
import {instanceOf, string, bool} from 'prop-types';
import {connect} from 'react-redux';
import {localization} from 'utils';

class LocaleDateTime extends PureComponent {
    static propTypes = {
        localeDateTime: instanceOf(Date).isRequired,
        locale: string.isRequired,
        className: string,
        spaceBefore: bool,
        spaceAfter: bool,
        indentBefore: bool,
        indentAfter: bool
    };

    static defaultProps = {
        className: '',
        spaceBefore: false,
        spaceAfter: false,
        indentBefore: false,
        indentAfter: false
    };

    render() {
        const {localeDateTime, locale, className, spaceBefore, spaceAfter, indentBefore, indentAfter} = this.props;
        const options = {
            day: 'numeric',
            month: 'short',
            year: 'numeric',
            hour: '2-digit',
            minute: '2-digit'
        };
        const spaceBeforeString = spaceBefore ? ' ' : '';
        const spaceAfterString = spaceAfter ? ' ' : '';
        const content = `${spaceBeforeString}${localization.getLocaleDateTime(localeDateTime, locale, options)}${spaceAfterString}`;
        const style = {
            paddingLeft: indentBefore ? '0.3125rem' : '',
            paddingRight: indentAfter ? '0.3125rem' : ''
        };
        return (
            <span className={className} style={style}>
                {content}
            </span>
        );
    }
}

const mapStateToProps = ({layoutReducer}) => ({
    locale: layoutReducer.locale
});

export default connect(mapStateToProps)(LocaleDateTime);
