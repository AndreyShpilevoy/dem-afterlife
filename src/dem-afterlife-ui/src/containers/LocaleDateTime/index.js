/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-this: 0 */

import React from "react";
import { instanceOf, string, bool } from "prop-types";
import { connect } from "react-redux";
import { localization } from "utils";
import { localeSelector } from "containers/selectors";

const LocaleDateTime = ({
  localeDateTime,
  locale,
  className,
  spaceBefore,
  spaceAfter,
  indentBefore,
  indentAfter
}) => {
  const options = {
    day: "numeric",
    month: "short",
    year: "numeric",
    hour: "2-digit",
    minute: "2-digit"
  };
  const spaceBeforeString = spaceBefore ? " " : "";
  const spaceAfterString = spaceAfter ? " " : "";
  const content = `${spaceBeforeString}${localization.getLocaleDateTime(
    localeDateTime,
    locale,
    options
  )}${spaceAfterString}`;
  const style = {
    paddingLeft: indentBefore ? "0.3125rem" : "",
    paddingRight: indentAfter ? "0.3125rem" : ""
  };
  return (
    <span className={className} style={style}>
      {content}
    </span>
  );
};

LocaleDateTime.propTypes = {
  localeDateTime: instanceOf(Date).isRequired,
  locale: string.isRequired,
  className: string,
  spaceBefore: bool,
  spaceAfter: bool,
  indentBefore: bool,
  indentAfter: bool
};

LocaleDateTime.defaultProps = {
  className: "",
  spaceBefore: false,
  spaceAfter: false,
  indentBefore: false,
  indentAfter: false
};

const mapStateToProps = state => ({
  locale: localeSelector(state)
});

export default connect(mapStateToProps)(LocaleDateTime);
