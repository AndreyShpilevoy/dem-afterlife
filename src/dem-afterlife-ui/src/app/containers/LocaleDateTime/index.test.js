/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 , react/jsx-filename-extension:0 */

import React from 'react';
import {mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import LocaleDateTime from './index';

describe('LocaleDateTime', () => {
    const mockStore = configureMockStore();

    it('component match expected snapshot for locale date', () => {
        const props = {
            store: mockStore({layoutReducer: {locale: 'ru'} }),
            localeDateTime: new Date('2017-06-02')
        };
        expect(mount(<LocaleDateTime {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for locale date with space before', () => {
        const props = {
            store: mockStore({layoutReducer: {locale: 'ru'} }),
            localeDateTime: new Date('2017-06-02'),
            spaceBefore: true
        };
        expect(mount(<LocaleDateTime {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for locale date with space after', () => {
        const props = {
            store: mockStore({layoutReducer: {locale: 'ru'} }),
            localeDateTime: new Date('2017-06-02'),
            spaceAfter: true
        };
        expect(mount(<LocaleDateTime {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });

    it('component match expected snapshot for locale date with space before and after', () => {
        const props = {
            store: mockStore({layoutReducer: {locale: 'ru'} }),
            localeDateTime: new Date('2017-06-02'),
            spaceBefore: true,
            spaceAfter: true
        };
        expect(mount(<LocaleDateTime {...props} />, {lifecycleExperimental: true})).toMatchSnapshot();
    });
});
