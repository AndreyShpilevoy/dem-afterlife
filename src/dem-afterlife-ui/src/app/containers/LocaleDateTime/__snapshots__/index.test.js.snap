// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleDateTime component match expected snapshot for locale date 1`] = `
<Connect(LocaleDateTime)
  localeDateTime={2017-06-02T00:00:00.000Z}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LocaleDateTime
    dispatch={[Function]}
    locale="ru"
    localeDateTime={2017-06-02T00:00:00.000Z}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      Jun 2, 2017, 3:00 AM
    </span>
  </LocaleDateTime>
</Connect(LocaleDateTime)>
`;

exports[`LocaleDateTime component match expected snapshot for locale date with space after 1`] = `
<Connect(LocaleDateTime)
  localeDateTime={2017-06-02T00:00:00.000Z}
  spaceAfter={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LocaleDateTime
    dispatch={[Function]}
    locale="ru"
    localeDateTime={2017-06-02T00:00:00.000Z}
    spaceAfter={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
      Jun 2, 2017, 3:00 AM 
    </span>
  </LocaleDateTime>
</Connect(LocaleDateTime)>
`;

exports[`LocaleDateTime component match expected snapshot for locale date with space before 1`] = `
<Connect(LocaleDateTime)
  localeDateTime={2017-06-02T00:00:00.000Z}
  spaceBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LocaleDateTime
    dispatch={[Function]}
    locale="ru"
    localeDateTime={2017-06-02T00:00:00.000Z}
    spaceBefore={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
       Jun 2, 2017, 3:00 AM
    </span>
  </LocaleDateTime>
</Connect(LocaleDateTime)>
`;

exports[`LocaleDateTime component match expected snapshot for locale date with space before and after 1`] = `
<Connect(LocaleDateTime)
  localeDateTime={2017-06-02T00:00:00.000Z}
  spaceAfter={true}
  spaceBefore={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LocaleDateTime
    dispatch={[Function]}
    locale="ru"
    localeDateTime={2017-06-02T00:00:00.000Z}
    spaceAfter={true}
    spaceBefore={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <span
      style={
        Object {
          "paddingLeft": "",
          "paddingRight": "",
        }
      }
    >
       Jun 2, 2017, 3:00 AM 
    </span>
  </LocaleDateTime>
</Connect(LocaleDateTime)>
`;
